apps/wallet/src/ui/app/hooks/deepbook/useGetEstimate.ts

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { type WalletSigner } from '_app/WalletSigner';
+import {
+	DEEPBOOK_KEY,
+	DEFAULT_WALLET_FEE_ADDRESS,
+	ESTIMATED_GAS_FEES_PERCENTAGE,
+	ONE_SUI_DEEPBOOK,
+	WALLET_FEES_PERCENTAGE,
+} from '_pages/swap/constants';
+import { useDeepBookContext } from '_shared/deepBook/context';
+import { FEATURES } from '_src/shared/experimentation/features';
+import { useFeatureValue } from '@growthbook/growthbook-react';
+import { useGetObject } from '@mysten/core';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { type DeepBookClient } from '@mysten/deepbook';
+import { TransactionBlock } from '@mysten/sui.js/builder';
+import { type CoinStruct, type SuiClient } from '@mysten/sui.js/client';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import BigNumber from 'bignumber.js';
+
+const MAX_COINS_PER_REQUEST = 10;
+
+async function getCoinsByBalance({
+	coinType,
+	balance,
+	suiClient,
+	address,
+}: {
+	coinType: string;
+	balance: string;
+	suiClient: SuiClient;
+	address: string;
+}) {
+	let cursor: string | undefined | null = null;
+	let currentBalance = 0n;
+	let hasNextPage = true;
+	const coins = [];
+	const bigIntBalance = BigInt(new BigNumber(balance).integerValue(BigNumber.ROUND_UP).toString());
+
+	while (currentBalance < bigIntBalance && hasNextPage) {
+		const { data, nextCursor } = await suiClient.getCoins({
+			owner: address,
+			coinType,
+			cursor,
+			limit: MAX_COINS_PER_REQUEST,
+		});
+
+		if (!data || !data.length) {
+			break;
+		}
+
+		for (const coin of data) {
+			currentBalance += BigInt(coin.balance);
+			coins.push(coin);
+
+			if (currentBalance >= bigIntBalance) {
+				break;
+			}
+		}
+
+		cursor = nextCursor;
+		hasNextPage = !!nextCursor;
+	}
+
+	if (!coins.length) {
+		throw new Error('No coins found in balance');
+	}
+
+	return coins;
+}
+
+function formatBalanceToLotSize(balance: string, lotSize: number) {
+	const balanceBigNumber = new BigNumber(balance);
+	const lotSizeBigNumber = new BigNumber(lotSize);
+	const remainder = balanceBigNumber.mod(lotSizeBigNumber);
+
+	if (remainder.isEqualTo(0)) {
+		return balanceBigNumber.toString();
+	}
+
+	const roundedDownBalance = balanceBigNumber.minus(remainder);
+	return roundedDownBalance.abs().toString();
+}
+
+async function getPlaceMarketOrderTxn({
+	deepBookClient,
+	poolId,
+	accountCapId,
+	address,
+	isAsk,
+	lotSize,
+	baseBalance,
+	quoteBalance,
+	quoteCoins,
+	walletFeeAddress,
+}: {
+	deepBookClient: DeepBookClient;
+	poolId: string;
+	accountCapId: string;
+	address: string;
+	isAsk: boolean;
+	lotSize: number;
+	baseBalance: string;
+	quoteBalance: string;
+	baseCoins: CoinStruct[];
+	quoteCoins: CoinStruct[];
+	walletFeeAddress: string;
+}) {
+	const txb = new TransactionBlock();
+	const accountCap = accountCapId || deepBookClient.createAccountCap(txb);
+
+	let swapCoin;
```

Comment:
do these need explicitly typed (e.g., are they typed as `any` down below)?

Reply:
removed this here and declare explicitly inside if/else